src/Facebook/FacebookClient.php

```diff
@@ -197,7 +197,7 @@ public function prepareRequestMessage(FacebookRequest $request)
      */
     public function sendRequest(FacebookRequest $request)
     {
-        if (get_class($request) === 'FacebookRequest') {
+        if (get_class($request) === 'Facebook\FacebookRequest') {
```

Comment:
```php
if (!$request instanceof FacebookBatchRequest) {
```
;)
@lexakozakov FacebookBatchRequest extends FacebookRequest, so you can pass a FacebookBatchRequest to sendRequest